recipe/2024-06-private-bicep-registries.md

```diff
@@ -0,0 +1,258 @@
+# Adding support for Private Bicep Registries
+
+* **Status**: Pending/Approved
+* **Author**: Vishwanath Hiremath (@vishwahiremat)
+
+## Overview
+
+Today, infrastructure recipes are restricted to being stored on publicly accessible OCI-compliant registries. This limitation means that Radius does not support the use of privately-hosted Bicep recipes, nor does it provide a mechanism to configure authentication for accessing these private registries. This gap is a significant hindrance for organizations that develop their own infrastructure modules and prefer to store them in secure, privately-accessible repositories.
+
+To address this issue and enhance the usability of Radius for serious enterprise use cases, we propose adding support for private Bicep registries. This feature will enable Radius to authenticate and interact with private OCI-compliant registries, ensuring secure access to proprietary bicep recipes. 
+
+## Terms and definitions
+
+| Term     | Definition                                                                                                                                                                                                 |
+| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| OCI compliant registry | An OCI-compliant registry refers to a container registry that adheres to the specifications set by the Open Container Initiative  |
+| ORAS | ORAS is the de facto tool for working with OCI Artifacts.(https://oras.land/) |
+| Private Bicep Registry | A private Bicep registry is a dedicated repository for storing and managing Bicep modules which is not accessible publicly need to authenticate into it for registry operations.| 
+
+## Objectives
+> **Issue Reference:** https://github.com/radius-project/radius/issues/6917
+
+### Goals
+- Add support to publish bicep recipes to a private bicep registry.
+- Enable support to use bicep recipes stored in private bicep registries.
+- Support all OCI compliant private registries.
+
+### Non goals
+
+- Workload identity based authentication(Will be covered as part of different design) 
```

Comment:
Good note on this one. Do we have clarity on what forms of authentication we will support?

Reply:
Here are the 2 forms of authentication we want to support:
1) Username-password based auth, (which i supported by most of the OCI complaint registries)
2) Federated authentication(Azure workload identity/ AWS IRSA) for ACR and ECR.

Reply:
Is 2. going to be a fast-follow after this? eg: are we going to enable username/password and then immediately enable AZWI/IRSA?

Reply:
Would be great to also call out why it will not be covered in initial support.

Reply:
I believe the plan of record is for workload and service identities to be in scope for the initial release.